packages/block-library/src/navigation/edit.js

```diff
@@ -102,6 +158,31 @@ function Navigation( {
 		}
 	);
 
+	// Turn on contrast checker for web only since it's not supported on mobile yet.
+	const enableContrastChecking = Platform.OS === 'web';
+
+	const [ detectedBackgroundColor, setDetectedBackgroundColor ] = useState();
+	const [ detectedColor, setDetectedColor ] = useState();
+	const [
+		detectedOverlayBackgroundColor,
+		setDetectedOverlayBackgroundColor,
+	] = useState();
+	const [ detectedOverlayColor, setDetectedOverlayColor ] = useState();
+
+	useEffect( () => {
+		if ( ! enableContrastChecking ) {
```

Comment:
This feels more like a block "supports" feature. https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/. Maybe @ajlende or @youknowriad might have thoughts.

Specifically something like:
```
supports: {
    color: {
        __experimentalContrastChecking: true,
    }
}
```

Is there any reason why we didn't update block.json to add supports: color. (color.background and color.text default to true on the object).

For dynamic blocks there are helpers in block-supports folder. https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/colors.php

Reply:
> This feels more like a block "supports" feature.

Do you mean `enableContrastChecking`? That's a variable that right now just checks if the platform is web, since mobile doesn't support contrast checking yet.

> Is there any reason why we didn't update block.json to add supports: color. (color.background and color.text default to true on the object).

This is more code that was copied from `packages/block-library/src/button/color-edit.js`, going off [this conversation](https://github.com/WordPress/gutenberg/issues/29963#issuecomment-823132744) to implement a customized version for this block instead of adding filters to the Color Panel.

Reply:
> Do you mean enableContrastChecking?

More that if folks find this feature useful, that others can enable it for other blocks without copy pasting.

> This is more code that was copied from packages/block-library/src/button/color-edit.js, going off this conversation to implement a customized version for this block instead of adding filters to the Color Panel.

Thanks for the extra context! Reading the conversation it sounds like we'd like to explore a bit and observe use cases. One thing that may be helpful for future us, is creating a follow up issue to articulate the concerns there.